dev/run-tests.py

```diff
@@ -0,0 +1,417 @@
+    checks to determine if this is running on an Amplab Jenkins machine"""
+
+    # base environment values for SBT_MAVEN_PROFILE_ARGS_ENV which will be appended on
+    sbt_maven_profile_args_base = ["-Pkinesis-asl"]
+
+    sbt_maven_profile_arg_dict = {
+        "hadoop1.0" : ["-Phadoop-1", "-Dhadoop.version=1.0.4"],
+        "hadoop2.0" : ["-Phadoop-1", "-Dhadoop.version=2.0.0-mr1-cdh4.1.1"],
+        "hadoop2.2" : ["-Pyarn", "-Phadoop-2.2"],
+        "hadoop2.3" : ["-Pyarn", "-Phadoop-2.3", "-Dhadoop.version=2.3.0"],
+    }
+
+    # set the SBT maven build profile argument environment variable and ensure
+    # we build against the right version of Hadoop
+    if os.environ.get("AMPLAB_JENKINS_BUILD_PROFILE"):
+        os.environ[SBT_MAVEN_PROFILE_ARGS_ENV] = \
+            " ".join(sbt_maven_profile_arg_dict.get(ajbp, []) 
+                     + sbt_maven_profile_args_base)
+    else:
+        os.environ[SBT_MAVEN_PROFILE_ARGS_ENV] = \
+            " ".join(sbt_maven_profile_arg_dict.get("hadoop2.3", [])
+                     + sbt_maven_profile_args_base)
+
+
+def is_exe(path):
+    """Check if a given path is an executable file
+    - from: http://stackoverflow.com/a/377028"""
+
+    return os.path.isfile(path) and os.access(path, os.X_OK)
+
+
+def which(program):
+    """Find and return the given program by its absolute path or 'None'
+    - from: http://stackoverflow.com/a/377028"""
+
+    fpath, fname = os.path.split(program)
+
+    if fpath:
+        if is_exe(program):
+            return program
+    else:
+        for path in os.environ.get("PATH").split(os.pathsep):
+            path = path.strip('"')
+            exe_file = os.path.join(path, program)
+            if is_exe(exe_file):
+                return exe_file
+    return None
+
+
+def determine_java_executable():
+    """Will return the *best* path possible for a 'java' executable or `None`"""
+
+    java_home = os.environ.get("JAVA_HOME")
+
+    # check if there is an executable at $JAVA_HOME/bin/java
+    java_exe = which(os.path.join(java_home, "bin/java"))
+    # if the java_exe wasn't set, check for a `java` version on the $PATH
+    return java_exe if java_exe else which("java")
+
+
+def determine_java_version(java_exe):
+    """Given a valid java executable will return its version in named tuple format
+    with accessors '.major', '.minor', '.patch', '.update'"""
+
+    raw_output = subprocess.check_output([java_exe, "-version"], 
+                                         stderr=subprocess.STDOUT)
+    raw_version_str = raw_output.split('\n')[0] # eg 'java version "1.8.0_25"'
+    version_str = raw_version_str.split()[-1].strip('"') # eg '1.8.0_25'
+    version, update = version_str.split('_') # eg ['1.8.0', '25']
+
+    JavaVersion = namedtuple('JavaVersion', 
+                             ['major', 'minor', 'patch', 'update'])
+
+    # map over the values and convert them to integers
+    version_info = map(lambda x: int(x), version.split('.') + [update])
```

Comment:
`pylint` also complains about this:

```
W:161,19: map/filter on lambda could be replaced by comprehension (deprecated-lambda)
W:161,23: Lambda may not be necessary (unnecessary-lambda)
```

You can just use a for-comprehension here instead.